vmdb/app/assets/javascripts/miq_application.js

```diff
@@ -298,33 +406,50 @@ function miqButtons(h_or_s, prefix) {
 
 // Hide/show form buttons
 function miqValidateButtons(h_or_s, prefix) {
-  var prefix = (prefix == null) ? "" : prefix;
-  on_id = prefix + 'validate_buttons_on';
-  off_id = prefix + 'validate_buttons_off';
-  if (miqDomElementExists('flash_msg_div')) $('flash_msg_div').hide();
-    if (h_or_s == "show") {
-      if(miqDomElementExists(on_id)) $("#" + on_id).show();
-      if(miqDomElementExists(off_id)) $("#" + off_id).hide();
-    } else {
-      if(miqDomElementExists(off_id)) $("#" + off_id).show();
-      if(miqDomElementExists(on_id)) $("#" + on_id).hide();
-}   }
+  var on_id = '#' + prefix + 'validate_buttons_on';
+  var off_id = '#' + prefix + 'validate_buttons_off';
+
+  if ($('#flash_msg_div').length) {
+    $('flash_msg_div').hide();
+  }
+  if (h_or_s == "show") {
+    if ($(on_id).length) {
+      $("#" + on_id).show();
+    }
+    if ($(off_id).length) {
+      $("#" + off_id).hide();
+    }
+  } else {
+    if ($(off_id).length) {
+      $("#" + off_id).show();
+    }
+    if ($(on_id).length) {
+      $("#" + on_id).hide();
+    }
+  }
+}
 
 // Convert Button image to hyperlink
 function toggleConvertButtonToLink(button, url, toggle) {
-  if (toggle == true) {
-    if(button.hasClass('dimmed')) {
-      button.removeClass('dimmed')
+  if (toggle) {
+    if (button.hasClass('dimmed')) {
+      button.removeClass('dimmed');
     }
-    if(button[0].parentNode.outerHTML.indexOf('<a href') == -1) {
-      button[0].outerHTML = "<a href=" + url + " title='" + button[0].getAttribute('alt') + "'>" + button[0].outerHTML + "</a>";
+    if (button[0].parentNode.outerHTML.indexOf('<a href') == -1) {
+      button[0].outerHTML =
+        "<a href=" +
+        url +
+        " title='" +
+        button[0].getAttribute('alt') +
+        "'>" +
+        button[0].outerHTML +
+        "</a>";
     }
-  }
-  else {
-    if(!button.hasClass('dimmed')) {
-      button.addClass('dimmed')
+  } else {
+    if (!button.hasClass('dimmed')) {
+      button.addClass('dimmed');
     }
-    if(button[0].parentNode.outerHTML.indexOf('<a href') > -1) {
+    if (button[0].parentNode.outerHTML.indexOf('<a href') > -1) {
```

Comment:
This ugly JS stuff can be replaced with `$()` selector. Something like this: `$(button).parent.is("a")`